test/extended/router/gatewayapicontroller.go

```diff
@@ -539,3 +591,74 @@ func assertHttpRouteSuccessful(oc *exutil.CLI, gwName, name string) (*gatewayapi
 	e2e.Logf("httpRoute %s/%s successful", namespace, name)
 	return checkHttpRoute, nil
 }
+
+// assertHttpRouteConnection checks if the http route of the given name replies successfully,
+// and returns an error if not
+func assertHttpRouteConnection(oc *exutil.CLI, hostname, gwName string) error {
+	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	ingressNameSpace := "openshift-ingress"
+	gateway, errGwStatus := gwapiClient.GatewayV1().Gateways(ingressNameSpace).Get(context.TODO(), gwName, metav1.GetOptions{})
+	if errGwStatus != nil || gateway == nil {
+		e2e.Failf("Unable to create httpRoute, no gateway available during route assertion %v", errGwStatus)
+	}
```

Comment:
Same as for the assertion of the dns record. This is redundant, when this function is called the gateway already exists and is accepted.